### ISSUE TITLE
Rust FFI constructor/destructor should use raw pointers

### ISSUE THREAD

**Issue by jdm** [dns2utf8/hour_glass#1]
Rather than returning and accepting `Box<IntSorter>`, we should make use of `Box::into_raw` and `Box::from_raw` to only pass around FFI-safe types instead.

**Comment by dns2utf8** [comment]
I implemented it with the `into_raw` and `from_raw` and it works just the same.

When I analize the difference it seems to me, that it returns the same pointer.

**Comment by jdm** [comment]
That is true today. It relies on the internal representation of `Box<T>`, however, which is not guaranteed to remain the same in the future.

**Comment by dns2utf8** [comment]
This is a good point, so it is ok to use the pointer directly like in this?: https://github.com/dns2utf8/hour_glass/blob/master/src/lib.rs#L35

**Comment by jdm** [comment]
The risk you face there is that the C code could pass a null pointer and you would end up with undefined behaviour, since LLVM optimizes code assuming those pointers could never be null.

**Comment by dns2utf8** [comment]
Ah, well that is ok. As long as the user uses C++ that case should never occure thanks to @fmorgner for implementing the facade with [std::unique_ptr](https://github.com/dns2utf8/hour_glass/blob/master/src/int_sorter_facade.h#L24)
